src/gebt_poc/mesh.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+namespace openturbine {
+class Mesh {
+public:
+    int GetNumberOfElements() { return number_of_elements_; }
+
+    int GetNumberOfNodes() { return number_of_nodes_; }
+
+    void SetNumberOfElements(int number_of_elements) { number_of_elements_ = number_of_elements; }
+    void SetNumberOfNodes(int number_of_nodes) { number_of_nodes_ = number_of_nodes; }
+
+    void InitializeElementNodeConnectivity(int number_of_elements, int nodes_per_element) {
+        connectivity_ = Kokkos::View<int**>("connectivity", number_of_elements, nodes_per_element);
+    }
+
+    KOKKOS_FUNCTION
+    Kokkos::View<const int*> GetNodesForElement(int elementID) {
+        return Kokkos::subview(connectivity_, elementID, Kokkos::ALL);
+    }
+
+    friend Mesh create1DMesh(int number_of_elements, int nodes_per_element);
+
+protected:
+    int number_of_elements_;
+    int number_of_nodes_;
+    Kokkos::View<int**> connectivity_;
+};
+
+inline Mesh create1DMesh(int number_of_elements, int nodes_per_element) {
+    Mesh mesh;
+    mesh.SetNumberOfElements(number_of_elements);
+    mesh.SetNumberOfNodes(number_of_elements * nodes_per_element - (number_of_elements - 1));
+    mesh.InitializeElementNodeConnectivity(number_of_elements, nodes_per_element);
+
+    auto set_element_connectivity = KOKKOS_LAMBDA(int elementID) {
+        for (int node = 0; node < nodes_per_element; ++node) {
+            mesh.connectivity_(elementID, node) = node + (nodes_per_element - 1) * (elementID);
+        }
+    };
+
+    Kokkos::parallel_for(number_of_elements, set_element_connectivity);
+    return mesh;
+}
```

Comment:
I might be missing something here, but I don't see why we need a separate class for creating 1D mesh. Couldn't we go through the interface of `Mesh` class and instantiate a 1D/2D/3D mesh as necessary?

Reply:
Since the user has to go through these functions, we can ensure that a valid mesh is created from a simple UI where opportunities to make an invalid mesh are encapsulated.  As we need different mesh topologies, we can add simple functions to make them or we can add functions that read them from other input sources. 

There are plenty of features that we could add to make things more flexible or different ways to do it, but I find this one both an simple solution to the problem at hand and easily extended to the problems in our immediate future.